code/modules/mob/living/silicon/silicon.dm

```diff
@@ -59,8 +59,9 @@
 	. = ..()
 	GLOB.silicon_mobs += src
 	faction += "silicon"
-	for(var/datum/atom_hud/data/diagnostic/diag_hud in GLOB.huds)
-		diag_hud.add_to_hud(src)
+	if(sensors_on)
```

Comment:
remove this check for sensors_on

this adds the pAI to the diag hud so it's visible to hud-wearers